libs/sdk-core/src/input_parser.rs

```diff
@@ -585,17 +585,24 @@ impl From<Uri<'_>> for BitcoinAddressData {
 }
 
 #[cfg(test)]
-mod tests {
+pub(crate) mod tests {
+    use std::sync::Mutex;
+
     use anyhow::anyhow;
     use anyhow::Result;
     use bitcoin::bech32;
     use bitcoin::bech32::{ToBase32, Variant};
     use bitcoin::secp256k1::{PublicKey, Secp256k1, SecretKey};
-    use mockito::Mock;
+    use mockito::{Mock, Server, ServerGuard};
+    use once_cell::sync::Lazy;
 
     use crate::input_parser::*;
     use crate::models::Network;
 
+    /// Mock server used in tests
+    pub(crate) static MOCK_HTTP_SERVER: Lazy<Mutex<ServerGuard>> =
+        Lazy::new(|| Mutex::new(Server::new()));
+
```

Comment:
I don't think this needs to be shared between tests. `let mut server = mockito::Server::new();` in the tests is sufficient and avoids shared state.

Reply:
That's what I started with, but the problem is mockito chooses a random port on start. We later need to know this port to replace "localhost" in test values with `localhost:port`. Not the best approach, I know, but I found no other way.

So this `static MOCK_HTTP_SERVER` is shared among tests to go around this issue with random ports.

Reply:
Ah it actually runs a real server? I didn't realize that.